screens/DetailsScreen/index.jsx

```diff
@@ -51,30 +59,86 @@ const styles = StyleSheet.create({
     color: 'white',
     fontSize: 16,
     fontWeight: '600',
-  }
+  },
+  errorText: {
+    textAlign: 'center',
+    marginTop: 20,
+    fontSize: 18,
+  },
+  favoriteButton: {
+    position: 'absolute',
+    top: 10,
+    right: 10,
+    padding: 8,
+    backgroundColor: 'rgba(255, 255, 255, 0.8)',
+    borderRadius: 30,
+    zIndex: 1,
+  },
+  favoriteIcon: {
+    fontSize: 24,
+    color: 'red',
+  },
 });
 
-const DetailsScreen = ({ route, navigation }) => {
+const DetailsScreen = ({ route, navigation, t }) => {
```

Comment:
+ add some checks, for cases when u won't recieve title, authors or other info from navigation options